lgc/patch/FragColorExport.cpp

```diff
@@ -99,224 +99,170 @@ namespace lgc {
 FragColorExport::FragColorExport(LLVMContext *context) : m_context(context) {
 }
 
+// =====================================================================================================================
+// Split a possible vector value, if there are less than 4 elements, append with null value.
+//
+// @param output : Fragment color output
+// @param builder : The IR builder for inserting instructions
+// @param [out] results : the returned elements
+static void splitVector(Value *input, BuilderBase &builder, Value *results[4]) {
+  Type *valueTy = input->getType();
+  unsigned compCount = valueTy->isVectorTy() ? cast<FixedVectorType>(valueTy)->getNumElements() : 1;
+
+  // This is used to handle scalar type
+  results[0] = input;
+
```

Comment:
```
if (!valueTy->isVectorTy())
  return
```
and remove the check from for loops

Reply:
we cannot return here. the scalar type also needs to be padded null values.

Reply:
You could fill it up front with with null values with something like `std::fill(results.being(), results.end(), Constant::getNullValue(valueTy->getScalarType())), since constants don't appear as instructions within functions in llvm and this won't cause any pollution. Or leave it as-is.